draft-robert-mimi-delivery-service.md

```diff
@@ -78,45 +78,49 @@ TODO: client identifiers definitions are preliminary.
 
 # Interfaces
 
-The MIMI DS tracks the state of individual MLS groups on a hub by processing
-group operations. The MIMI DS can process the following requests originating
-from clients:
-
-* Add clients to a group
-* Inject the sending client into a group (external join)
-* Remove clients from a group
-* Updates their MLS key material
-* Send application messages
+The MIMI DS protocol inherits the proposal-commit logic from MLS. Any party that
+is either a group member, or that as added as an external sender can propose
+changes to the group such as the addition, removal, or update of clients. Group
+members can then change the group state in a commit operation. A commit MUST
+include all valid previously received proposals, as well as any valid proposals
+that the committing client wants to add to the commit.
+
+The MIMI DS tracks the state of individual MLS groups on a hub by storing
+proposals and processing commits.
+
+Any party (including group members and external senders) can send requests for
+the current group information (an MLS GroupInfo, required for clients to join
+the group without the help of a group member) or request key material (MLS
+KeyPackages, required by group members to add new group members) of the DS'
+clients.
 
-TODO: This list will either grow longer if we want to keep distinct operations,
-or shorter, if we don't. In the latter case, we'll need more extensive commit
-validation.
+Group members and external senders can send send proposals to the group. Valid
+proposals include any proposals in {{!RFC9420}}:
 
-Follower servers can send requests the following:
+* Add proposals (to add new group members)
+* Remove proposals (to remove existing group members)
+* Update proposals (to update a group members's key material)
+* PSK proposals (to inject additional key material into the group)
+* Re-Init proposals (to re-initialize the group, e.g. with a new ciphersuite)
+* Group Context Extensions (to modify data in GroupContext Extensions, e.g. to
+  add/remove/update external senders)
```

Comment:
Don't we want to allow any kind of proposal as long as it is supported by the group?

Reply:
Good point. I mostly wanted to give an overview over the operations that are generally possible. But of course, the MIMI DS protocol itself does not impose any particular restrictions.